controllers/mediaController.js

```diff
@@ -0,0 +1,87 @@
+// import Media schema from models
+const {User} = require('../models/mainModels.js');
+
+// object contains all mediaController middleware
+const mediaController = {};
+
+// middleware to get entire media profile of user
+mediaController.getMedia = (req, res, next) => {
+  // find all media catalog
+  User.findOne({_id: req.query.userId})
+    .exec()
+    // response is the found User
+    .then((response) => {
+      // console.log to see if fired
+      console.log('mediaController getMedia fired');
+      // if no user found
+      if (!response) {
+        // store an empty object in local memory for front end
+        res.locals.media = {};
+        return next();
+      }
+      // store in local memory the User.media
+      res.locals.media = response.media;
+      return next();
+    })
+    .catch((error) => {
+      console.log(err.stack);
+      return next('error in getMedia middleware');
+    });
+};
+
+// A possible approach (syntax is not exactly correct - this is the general idea) for add media into the array
+
+// User.findOne({_id: req.params.userId}).
+// then -> userMedia = user.media
+// userMedia.push(req.body)
+
+// User.update({_id: req.params.userId}, {media: userMedia})
+
+// add a media type to user profile
+mediaController.addMedia = (req, res, next) => {
+  // finds and update media property on user and returns the updated user
+  User.findOneAndUpdate(
+    // filter for the _id
+    {_id: req.params.userId},
+    // $addToSet is a mongoDb method to add into media Array without duplicates
+    {$addToSet: {media: userMedia}}
```

Comment:
what is "userMedia"? it seems like it is not defined in this middleware. 

Additionally, the response should return the added media document in the body